app/scripts/components/common/map/controls/aoi/reset-aoi-control.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useCallback } from 'react';
+import { CollecticonArrowLoop } from '@devseed-ui/collecticons';
+import { Button, createButtonStyles } from '@devseed-ui/button';
+import styled from 'styled-components';
+import { themeVal } from '@devseed-ui/theme-provider';
+import useThemedControl from '../hooks/use-themed-control';
+import useMaps from '../../hooks/use-maps';
+
+const SelectorButton = styled(Button)`
+  &&& {
+    ${createButtonStyles({ variation: 'primary-fill', fitting: 'skinny' })}
+    background-color: ${themeVal('color.surface')};
+    &:hover {
+      background-color: ${themeVal('color.surface')};
+    }
+    & path {
+      fill: ${themeVal('color.base')};
+    }
+  }
+`;
+
+function ResetAoI() {
+  // This doesn't work as this Context is not reachable frome here :()
+  const { main } = useMaps();
+
+  const onReset = useCallback(() => {
+    const mbDraw = (main as any)?.instance?._drawControl;
+    if (!mbDraw) return;
+    mbDraw.trash();
+  }, [main]);
+
+  return (
+    <>
+      <SelectorButton onClick={onReset}>
+        <CollecticonArrowLoop title='Upload geoJSON' meaningful />
+      </SelectorButton>
+    </>
+  );
+}
+
+export default function ResetAoIControl() {
+  useThemedControl(
+    () => {
+      return <ResetAoI />;
+    },
+    {
+      position: 'top-left'
+    }
+  );
+  return null;
+}
```

Comment:
Would it be possible to access the context here and pass it as a prop to `ResetAoI`?

Reply:
Great suggestion: https://github.com/NASA-IMPACT/veda-ui/pull/707/commits/d102d801642e909b69ea3f88ea16b04a9559a29b

However `instance` doesn't exist on this map ref for some reason.

I could create a specific context for this map implementation in order for the draw control to expose its internals for other draw-related controls, but this really feels like a hack.

Reply:
Where is `instance` coming from?

Reply:
The previous `ref` implementation had an `instance` because we added it since it served our needs.

https://github.com/NASA-IMPACT/veda-ui/blob/1bb5c5efe94767c0b4e3a55387e8e22c024119db/app/scripts/components/common/mapbox/index.tsx#L168-L175

And then the map stored the draw control so we can act on it, but none of this is default.

https://github.com/NASA-IMPACT/veda-ui/blob/1bb5c5efe94767c0b4e3a55387e8e22c024119db/app/scripts/components/common/mapbox/aoi/mb-aoi-draw.js#L32-L40

Reply:
A solution for this implementation could be to add a reference to the control when it is mounted:

`@app/scripts/components/common/map/controls/aoi/index.tsx`
```diff
useControl<MapboxDraw>(
-    () => {
+    ({map}) => {
      control.current = new MapboxDraw(props);
+      map._drawControl = control.current;
      return control.current;
    },
```

Then you could access it via `map._drawControl` on the reset button

Reply:
However for the clear button I don't think you need to access the `trash` method. Using this without a selection does nothing.
I'd imagine that deleting all the feature via the atom would do the trick.

Reply:
I've used `deleteAll`: https://github.com/NASA-IMPACT/veda-ui/pull/707/commits/d102d801642e909b69ea3f88ea16b04a9559a29b, it works!
> I'd imagine that deleting all the feature via the atom would do the trick.
Sadly no, you have to manually keep the atom and the mbdraw state in sync

I have taken the opportunity to clean up that flow with the upload AoI feature: https://github.com/NASA-IMPACT/veda-ui/pull/707/commits/66a9f32ebcee06e41f3bf7ae8da1f16f31c0b024